.github/scripts/update.py

```diff
@@ -208,9 +208,12 @@ def write_commit_msg(args, new_files: int, updated_files: int):
         cfg_automation, args.branch, args.firefox, args.configs
     )
 
-    if cfg_automation["paths"] != source_dirs and args.branch == cfg_automation["head"]:
-        # Write back updated configuration
-        cfg_automation["paths"] = source_dirs
+    if cfg_automation["paths"] != source_dirs:
+        # Write back updated configuration, making sure that the list of paths
+        # is a superset of the configuration across all branches.
+        cfg_automation["paths"] = sorted(
+            list(set(cfg_automation["paths"] + source_dirs))
+        )
```

Comment:
This won't only be a superset across all branches, but also all of their previous states -- there is no facility here to prune the list of paths, only expand it. But that's probably fine? We can handle that manually with an update maybe once a year or so.

Reply:
Good point. That's probably something we need to document as a step when removing ESR branches, which we haven't done yet.